addons/note/models/note.py

```diff
@@ -82,6 +83,10 @@ def name_create(self, name):
     @api.model
     def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True):
         if groupby and groupby[0] == "stage_id":
+            if len(groupby) > 1 and not lazy:
+                # the implementation below does not handle this case
+                raise UserError(_("Grouping notes by %s is not possible.") % ", ".join(groupby))
```

Comment:
Maybe extend this comment for the future? While the comment is correct, it's not obvious that it's mainly related to read_progress_bar. Also, the comment could describe the problem behind implementing such case:

> for progressbar `groupby` is equal by default to `['stage_id', 'activity_state']`, where  `acitvity_state` is computed too. So, to properly handle this case, we need to mix the code below with `_read_progress_bar` from addons/mail/models/mail_activity.py, which would be too comlicated